### PR TITLE
Update colour picker

### PR DESCRIPTION
**Pull request by loganDoran**
#### Things done:
- Update colour picker section of sidebar to accommodate multiple colour inputs. 
    - Sidebar now handles either second colour input or calculates a range of colours based on primary colour selection.
- Correct map colour logic
    - Bug was found where primary colour wasn’t included in data colour gradient. This was fixed.
- Add toggle component
    - Toggle button to switch between methods of map colour generation.

#### Things to consider:
- UI of colour picker could be redone to make more intuitive. 
- Add info tool tip on how colour picker works?

### CODE REVIEW

File: client/src/components/geoJSONMap.tsx [hunk 1]
```diff
@@ -30,16 +30,17 @@ const GeoJSONMap: React.FC<GeoJSONMapProps> = ({ geoJsonData }) => {
...
+    const {primaryColorPicker, secondaryColorPicker, featureVisibility, autoColourRange} = useToggle();
 
     // Effect to initialize color gradient and data values
     useEffect(() => {
         if (geoJsonData) {
             setValues(extractValuesFromGeoJSON(geoJsonData));
-            const rgbColor = hexToRgb(colorPickerColor);
-            setColorGradient(generateColorGradient(steps, rgbColor));
+            const rgbColor = hexToRgb(primaryColorPicker);
+            const rgbColor_2 = hexToRgb(secondaryColorPicker);
```
Comment (ishraqn) [inline client/src/components/geoJSONMap.tsx:40]: the var names could be more verbose.

File: client/src/components/sidebar.css [hunk 2]
```diff
@@ -129,12 +130,43 @@
     cursor: pointer;
 }
 
+.toggle-button {
```
Comment (ishraqn) [inline client/src/components/sidebar.css:133]: the name could be more related to the color picker.

File: client/src/components/sidebar.tsx [hunk 3]
```diff
@@ -170,24 +179,41 @@ const Sidebar: React.FC<SidebarProps> = ({handleDownload, geoJsonData}) => {
...
-									{group.name}
-								</label><ColorPickerComponent
-										onColorChange={(colorResult) => setColorPickerColor(colorResult.hex)} /></>
+								<>
+									<>
+										<label className="menu-item-label">
+											{group.name}
+										</label>
+										<button
+											className="toggle-button"
```
Comment (ishraqn) [inline client/src/components/sidebar.tsx:188]: same as before, the name could be more related to color picker.

File: client/src/components/geoJSONMap.tsx (hunk 1 above)
Comment (loganDoran) [reply client/src/components/geoJSONMap.tsx:40]: So fair, think I made these just as placeholders to test then forgot to change them.

**Pull request merged**